File: Medium/0209_minimum-size-subarray-sum/Solution.py
class Solution:
    def minSubArrayLen(self, target: int, nums: list[int]) -> int:
        left = 0
        right = 0
        sum_val = 0
        min_len = float('inf')

        while right < len(nums):
            sum_val += nums[right]

            while sum_val >= target:
                if min_len > (right - left + 1):
                    min_len = right - left + 1

                sum_val -= nums[left]
                left += 1
            right += 1

        if min_len == float('inf'):
            return 0

        return min_len

File: Medium/0209_minimum-size-subarray-sum/Solution.cpp
#include <vector>
#include <climits>

class Solution {
public:
    int minSubArrayLen(int target, std::vector<int>& nums) {
        int left = 0;
        int right = 0;
        int sum = 0;
        int min_len = INT_MAX;

        while (right < nums.size()) {
            sum += nums[right];

            while (sum >= target) {
                if (min_len > (right - left + 1)) {
                    min_len = right - left + 1;
                }

                sum -= nums[left];
                left++;
            }
            right++;
        }

        if (min_len == INT_MAX) {
            return 0;
        }

        return min_len;
    }
};